daft/table/table_io.py

```diff
@@ -398,6 +399,42 @@ def read_csv(
     return _cast_table_to_schema(daft_table, read_options=read_options, schema=schema)
 
 
+@dataclass
+class _TableWriteData:
+    table: MicroPartition
+    path: str
+    partition_values: dict[str, Any]
```

Comment:
We can build a better abstraction here, such as a `TablePartitionValues` which can contain a micropartition

* For hive writes, you would retrieve a path using `TablePartitionValues.get_path()`
* For Iceberg, you can retrieve a path AND also iceberg-compatible partition values (by calling the functionality in `to_partition_representation`)
* Similar for delta.